### PR TITLE
Added Azure Function to ingest data reports via http

### PR DESCRIPTION
**Pull request by jimduff-usds**
This PR does ...

## Changes
- Added the new Report ingest Azure function
- added a few dependencies to pom.xml

## Checklist
- [X ] Tested locally?
- [X ] Added new dependencies?
- [ ] Updated the docs?
- [ ] Added a test?
- [ X] Did you check for sensitive data, and remove any?
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

### CODE REVIEW

**Comment by jimduff-usds** [comment]
Issue #71

File: prime-router/src/main/kotlin/azure/IngestFunction.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        context: ExecutionContext
+    ): HttpResponseMessage {
+        context.logger.info("report ingest function received an upload request.")
+
+        // Create an object representing the incoming CVS file, and validate presence of required metadata.
+        val file = try {
+            IngestedFile(request, context)
+        } catch (e: Exception) {
+            val msgs = TextStringBuilder()
+            Arrays.asList(*e.suppressed).forEach(Consumer { ex: Throwable -> msgs.appendln(ex.message) })
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/azure/IngestFunction.kt:47]: Could this be better expressed as 
```
e.suppressedExceptions.forEach{ msgs.appendln(it.message) }
```

**Review by RickHawesUSDS — APPROVED** [review]
Looks good. Made a few code style suggestions which you take or get to later.

File: prime-router/src/main/kotlin/azure/IngestedFile.kt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            errors.add("Error:  Missing schemaName parameter")
+        }
+        // @todo Should we allow an empty report data file, or is that always an error?
+        if (StringUtils.isEmpty(reportContent)) {
+            errors.add("Error:  incoming data file is empty or missing")
+        }
+        if (!errors.isEmpty()) {
+            val e = Exception()
+            for (error in errors) {
+                e.addSuppressed(Exception(error))
```
[the remark this answers is not in the document]
Comment (jimduff-usds) [reply prime-router/src/main/kotlin/azure/IngestedFile.kt:69]: Good catch - this was a leftover from the java translation.

**Comment by jimduff-usds** [comment]
Thanks, these are helpful tips; I'll make the changes in the next commit.

File: prime-router/src/main/kotlin/azure/IngestFunction.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+     *    {"filename":"lab1-test_results-17-42-31.csv","topic":"covid-19","schemaName":"pdi-covid-19.schema","action":"","blobURL":"http://azurite:10000/devstoreaccount1/ingested/lab1-test_results-17-42-31-pdi-covid-19.schema-3ddef736-55e1-4a45-ac41-f74086aaa654.csv"}
+     */
+    @FunctionName("report")
+    @StorageAccount("AzureWebJobsStorage")
+    fun run(
+        @HttpTrigger(
+            name = "req",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.FUNCTION
+        ) request: HttpRequestMessage<String?>,
```
[the remark this answers is not in the document]
Comment (jimduff-usds) [reply prime-router/src/main/kotlin/azure/IngestFunction.kt:37]: Haven't tested this in the cloud yet, but when run locally, it automatically uses ANONYMOUS, which is good.

**Pull request merged**